api/apikeys.go

```diff
@@ -58,3 +158,47 @@ func (api *API) userAPIKeyDELETE(u *database.User, w http.ResponseWriter, req *h
 	}
 	api.WriteSuccess(w)
 }
+
+// userAPIKeyPUT updates an API key. Only possible for public API keys.
+func (api *API) userAPIKeyPUT(u *database.User, w http.ResponseWriter, req *http.Request, ps httprouter.Params) {
+	akID, err := primitive.ObjectIDFromHex(ps.ByName("id"))
+	if err != nil {
+		api.WriteError(w, err, http.StatusBadRequest)
+		return
+	}
+	var body APIKeyPUT
+	err = parseRequestBodyJSON(req.Body, LimitBodySizeLarge, &body)
+	if err != nil {
+		api.WriteError(w, err, http.StatusBadRequest)
+		return
+	}
+	err = api.staticDB.APIKeyUpdate(req.Context(), *u, akID, body.Skylinks)
```

Comment:
Because the method doesn't take a pointer.

The general rule is that you pass a reference when:
a) you want to modify the value
b) the value is nested and working with a copy might create concurrency issues (e.g. copying mutexes, etc.)
c) the value is large and copying it will incur a performance hit

Neither of those is present here.

Reply:
> Because the method doesn't take a pointer.

this made me laugh 😁